zephyr/CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2022 Legrand North America, LLC.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+
+cmake_minimum_required(VERSION 3.10 FATAL_ERROR)
+
+
+# Detect the platform reliably
+if(ZEPHYR_BASE)
+else()
+  message(FATAL_ERROR "ZEPHYR_BASE needs to be defined for Zephyr builds")
+endif()
+
+# Verify the module name is set correctly
+set(expected_module_name "mcuboot")
+if(NOT "${module_name}" STREQUAL ${expected_module_name})
+  message(FATAL_ERROR "A module configuration error has occurred.
+This module is named '${module_name}' when '${expected_module_name}' \
+is expected. 
+Verify `module.yml` contains the line 'name: ${expected_module_name}'.")
+endif()
+unset(expected_module_name)
+
+#Do not allow in source builds
```

Comment:
Missing space after #.

Reply:
Since this is a brand new file I am fine with fixing formatting. Has the MCUboot project defined a source file formatting standard or policy?

Reply:
Fair point, given this is a new file.  I'll clean it up shortly.